### DISCUSSION TITLE
PT-710BT on Windows

### DISCUSSION THREAD

**Discussion by rpedde** [rpedde/pt750#94] [Q&A]
(Converted from issue)

**Comment by GC-RnD** [comment]
I have two brother label printers...
Brother PT-P710BT new CUBE+  and older Brother PT-2730
I am using win32print for usb connection and have a static label design.
The **_Brother PT-2730 works great,_** but the Brother PT-P710BT has spooler hang after the label is printed and cut.
The spooler thinks their are more pages...
![image](https://github.com/rpedde/pt750/assets/18072415/1128424f-be6d-412a-88e9-0aa9b358345f)
Job 3 ???, 1of 0 pages ???
I saw one comment that thinks the Bluetooth might be the cause.
Possibly you could test this on your label printer, using 12mm laminate.
Any suggestions or help is greatly appreciated !!

```
from PIL import Image, ImageFont, ImageDraw
from win32 import win32print

printer_name = "Brother PT-P710BT"  # printer_name = "Brother PT-2730"

def print_out():
	image = Image.new(mode="1", size=(160, 128), color=1)
	draw = ImageDraw.Draw( image )
	font = ImageFont.truetype('arialbd.ttf', 12)
	label_text = f'Test-Me (123) 456-7890\n Report #:  rpt\n       SN #:  ser\n    Issued:  cal_date\n Cal. Due:  due_date'
	draw.text((0, 28), label_text, font=font)

	image = image.transpose(Image.Transpose.ROTATE_90)
	image = image.transpose(Image.Transpose.FLIP_TOP_BOTTOM)
	image_data = bytearray(image.tobytes())

	for idx in range(len(image_data)):
			image_data[idx] = image_data[idx] ^ 0xFF

	bytes = b"\x00" * 100          # reset stream
	bytes += b"\x1B\x40"            # inittialize
	bytes += b"\x1b\x69\x61\x01"    # switch dynamic command mode, raster mode
	bytes += b"\x1B\x69\x21\x01"    # status notification mode, notify 
	bytes += b"\x1B\x69\x7A\x84\x00\x0C\x00\xa0\x00\x00\x00\x00\x00" # Set media and quality 
	bytes += b"\x1b\x69\x4d\x40"    # various mode settings, biary number... auto cut on x40 no mirror
	bytes += b"\x1b\x69\x4b\x08"    # advanced mode settings, 0000 1000 = 08  1000 1000 = 88
	bytes += b"\x1B\x69\x64\x0e\x00"# 2mm margin 0e = 14 dots
	# bytes += b"\x1B\x69\x61\x01") # specify the page number in “cut each * labels”
	bytes += b"\x4d\x02"            # compression

	for line in range(image.height):
		byte_ofs = 16 * line
		row_bytes = image_data[byte_ofs : byte_ofs + 16]
		print('row_bytes ', row_bytes)

		# raster graphics transfer
		bytes += b"\x47\x11\x00\x0F"    # "compressed" 16 literal bytes  47 02 00 F1 00 b"\x47\x11\x00\x0F"
		bytes += row_bytes

	bytes +=b"\x1a"                    # print and feed

	p = win32print.OpenPrinter(printer_name)
	win32print.StartDocPrinter(p, 1, ("print label", None, "raw"))
	win32print.StartPagePrinter(p)
	win32print.WritePrinter(p, bytes)  # send
	win32print.EndPagePrinter(p)
	win32print.EndDocPrinter(p)
	win32print.ClosePrinter(p)

print_out()
```


Don't know why the code is not formatted correctly.

**Comment by rpedde** [comment]
I have a PT-P710BT on order.  I'll mess with it a bit and see if I can get anything going.  I don't have a ton of windows experience, but I'll give it a throw.

**Comment by GC-RnD** [comment]
That great Ron…

I wrote an elaborate program for our Metrology lab that creates pdf certificates and labels to be affixed to calibrated items.
Spent more time trying to understand these PT label makers.
You code was the cleanest out of dozens of other projects.

I hope you have better luck than I !!!

**Reply by rpedde** [reply]
This worked okay for me.  I used your code as-is with python.org installed python 3.11.3 on a windows 11.  I installed the printer as generic/text only on USB001 (which is what it came up at), and it spit out your metro tag with no issues.  (other than REALLY SMALL, lol).

I ran the code in a venv with nothing but pillow installed and the default python.

Happy to screenshot stuff or something if it's helpful.

**Comment by GC-RnD** [comment]
Wow that was quick !!!
So if code works for you… then my win 11 is the issue.
Reinstalled printer driver before reaching out to you, the same issue.

Next attempted is setting up a virtual environment on my laptop and give it a try.
I need a simple remote development system.

I let you know how it goes.
Thank you for the quick response !!!

**Reply by rpedde** [reply]
No problem.  Yell if I can help, since I seem to have a working system for comparison.

**Comment by GC-RnD** [comment]
So you never installed the Brother PT-P710BT **_DRIVER_**...
but created a Generic / Text driver on the usb port showing PT-P710BT.
I removed the PT-P710BT driver, created a Generic / Text driver, and all is good !!!!
So the Brother PT-P710BT has some issues being driven from win32print with Brother driver,
but not from P-Touch editor software or a Generic / Text driver.

This will get us by for now...
but in the future if a new printer is purchased,
no one's going to know what to do.

Thanks Ron for teaching me something new !!!
